ui/app/mainui/ToastsManager.qml

```diff
@@ -0,0 +1,160 @@
+import QtQuick 2.15
+
+import utils 1.0
+
+import AppLayouts.stores 1.0
+import AppLayouts.Chat.stores 1.0 as ChatStores
+
+import shared.stores 1.0 as SharedStores
+
+// The purpose of this class is to be the central point for generating toasts in the application.
+// It will have as input all needed stores.
+// In case the file grows considerably, consider creating different toasts managers per topic / context
+// and just instantiate them in here.
+QtObject {
+    id: root
+
+    // Here there are defined some specific actions needed by a toast.
+    // They are normally specific navigations or open popup action.
+    enum ActionType {
+        None = 0,
+        NavigateToCommunityAdmin = 1,
+        OpenFinaliseOwnershipPopup = 2,
+        OpenSendModalPopup = 3
+    }
+
+    // Stores:
+    required property RootStore rootStore
+    required property ChatStores.RootStore rootChatStore
+    required property SharedStores.CommunityTokensStore communityTokensStore
+
+    // Properties:
+    required property var sendModalPopup
+
+    // Utils:
+    readonly property string viewOptimismExplorerText: qsTr("View on Optimism Explorer")
```

Comment:
We should have a generic version for different chains.

Reply:
Ok! Whenever other chains are used, we can easily parametrize.

Reply:
Then `onSendOwnerTokenStateChanged` and `onSetSignerStateChanged` should give us the `chainId`.